### PR TITLE
Blaze: Fix issue mapping campaign list

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #11526

## Description

This PR fixes the mapping issue for campaign list due to the change in API response:
- Updated the `BlazeCampaign` model: 
  - Replaced the mapping of `target_urn` with `target_url` and replaced property `productID` with `productURL`
  - Replaced the mapping of `total_budget` with `budget_cents` and renamed the property `totalBudget` to `budgetCents`.
- Updated the conversion between storage and read-only models. We must update the storage model to use new properties in the next app version.
- Updated the Blaze campaign item to display budget by dividing cents by 100.
- Updated the check for product with existing campaigns by checking for product URL instead of product ID.

## Testing instructions

- Log in to a store that is eligible for Blaze.
- Create a new Blaze campaign if needed.
- Open the campaign list, make sure that the list is correct.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Model/Storage/BlazeCampaign+ReadOnlyConvertible.swift [hunk 1]
```diff
@@ -24,13 +24,13 @@ extension Storage.BlazeCampaign: ReadOnlyConvertible {
     public func toReadOnly() -> BlazeCampaign {
         BlazeCampaign(siteID: siteID,
                       campaignID: campaignID,
-                      productID: productID?.int64Value,
+                      productURL: "", // TODO: map the new attribute `productURL` here
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Model/Storage/BlazeCampaign+ReadOnlyConvertible.swift]: nit: Can we create a GH issue to track these storage layer changes and add the issue number to the TODO?

File: WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignItemView.swift [hunk 2]
```diff
@@ -87,7 +87,7 @@ struct BlazeCampaignItemView: View {
                 VStack(alignment: .leading, spacing: Layout.statsVerticalSpacing) {
                     Text(Localization.budget)
                         .subheadlineStyle()
-                    Text(String(format: "%.2f", campaign.totalBudget))
+                    Text(String(format: "%.0f", campaign.budgetCents/Constants.centsToUnit))
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignItemView.swift]:
```suggestion
                    Text(String(format: "%.0f", campaign.budgetCents / Constants.centsToUnit))
```

File: Networking/NetworkingTests/Mapper/BlazeCampaignListMapperTests.swift [hunk 3]
```diff
@@ -16,12 +16,12 @@ final class BlazeCampaignListMapperTests: XCTestCase {
         let item = try XCTUnwrap(campaigns.first)
         XCTAssertEqual(item.siteID, dummySiteID)
         XCTAssertEqual(item.campaignID, 34518)
-        XCTAssertEqual(item.productID, 134)
```
Comment (selanthiraiyan) [inline Networking/NetworkingTests/Mapper/BlazeCampaignListMapperTests.swift:19]: nit: We should remove the `target_urn` field from the "blaze-campaigns-success" mock JSON file.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for working on this, Huong! 🎉 

I left a few non-blocking nits.

File: Yosemite/Yosemite/Model/Storage/BlazeCampaign+ReadOnlyConvertible.swift (hunk 1 above)
Comment (itsmeichigo) [reply Yosemite/Yosemite/Model/Storage/BlazeCampaign+ReadOnlyConvertible.swift]: Updated in bed4ed901465cc2f0b0eec808056b3cf776492bd.

File: WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignItemView.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignItemView.swift]: Updated in e6a6617dac9c66ffdfc6ff93ea4ea72c603f91b7.

File: Networking/NetworkingTests/Mapper/BlazeCampaignListMapperTests.swift (hunk 3 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/BlazeCampaignListMapperTests.swift:19]: Updated in a4647962ecc5c3a2ba47fd63fb523e75c4c0b011.

**Comment by itsmeichigo** [comment]
@spencertransier Please also include this in the final build of 16.7, thank you!

**Pull request merged**